types/enzyme/index.d.ts

```diff
@@ -372,17 +374,17 @@ export class ShallowWrapper<P = {}, S = {}, C = Component> {
      * Find every node in the render tree that matches the provided selector.
      * @param selector The selector to match.
      */
-    find<P2>(component: ComponentClass<P2>): ShallowWrapper<P2, any>;
     find<P2>(statelessComponent: StatelessComponent<P2>): ShallowWrapper<P2, never>;
+    find<P2>(component: ComponentType<P2>): ShallowWrapper<P2, any>;
```

Comment:
I don't understand why `StatelessComponent` needs to be present in this overload (via the union). It should be matched by what is now the first overload. Can you give an example of what was broken before that this change fixes?

Reply:
@jwbay Yeah sure.

When you use HOC typed like that
```
function withNamespaces(): <P extends WithNamespaces>(
    component: React.ComponentType<P>,
  ) => React.ComponentType<Subtract<P, WithNamespaces>>
``` 

The class lose his type for ComponentType = Class | Stateless
```
class MyComponent extend Component<Props>
export withNamespaces()(MyComponent)
```

Then, when you use
```
mount(Something).find<Props>(MyComponent)
```

My Component is not a Stateless because it's Stateless OR Class, so it doesnt match the Stateless case.

Reply:
I see, ty. 👍 Could you add tests to prevent us from regressing this fix?